src/gb/gb.c

```diff
@@ -420,6 +420,12 @@ void GBReset(struct LR35902Core* cpu) {
 				memcpy(&gb->memory.romBase[0x100], &gb->memory.rom[0x100], sizeof(struct GBCartridge));
 			}
 
+			if (gb->model == GB_MODEL_AGB) {
```

Comment:
I'm not very fond of having a hack to do this, but I'll let you know if I come up with a better idea.